packages/js/msgpack/src/index.ts

```diff
@@ -20,16 +22,23 @@ extensionCodec.register({
     if (object instanceof Map) {
       const optimized: Record<string | number, unknown> = {};
       for (const [key, value] of object) {
-        optimized[key] = value;
+        optimized[key] = msgpackEncode(value);
       }
-      return encode(optimized);
+      return msgpackEncode(optimized);
     } else {
       return null;
     }
   },
   decode: (data: Uint8Array) => {
-    const map = decode(data) as Record<string | number, unknown>;
-    return new Map(Object.entries(map));
+    const obj = msgpackDecode(data) as Record<string | number, unknown>;
+    const map = new Map();
+
+    for (const [key, value] of Object.entries(obj)) {
+      if (isBuffer(value)) {
```

Comment:
What if it's a key => buffer map?

Reply:
Nevermind it was unnecessary in the first place so removed it